ouroboros-consensus/src/Ouroboros/Consensus/Ledger/Tables.hs

```diff
@@ -0,0 +1,551 @@
+    -> LedgerTables l mk1
+    -> LedgerTables l mk2
+
+  traverseLedgerTables ::
+       Applicative f
+    => (forall k v .
+           (Ord k, Eq v)
+        =>    mk1 k v
+        -> f (mk2 k v)
+       )
+    ->    LedgerTables l mk1
+    -> f (LedgerTables l mk2)
+
+  zipLedgerTables ::
+       (forall k v.
+            (Ord k, Eq v)
+         => mk1 k v
+         -> mk2 k v
+         -> mk3 k v
+       )
+    -> LedgerTables l mk1
+    -> LedgerTables l mk2
+    -> LedgerTables l mk3
+
+  zipLedgerTables2 ::
+       (forall k v.
+            (Ord k, Eq v)
+         => mk1 k v
+         -> mk2 k v
+         -> mk3 k v
+         -> mk4 k v
+       )
+    -> LedgerTables l mk1
+    -> LedgerTables l mk2
+    -> LedgerTables l mk3
+    -> LedgerTables l mk4
+
+  zipLedgerTablesA ::
+       Applicative f
+    => (forall k v.
+            (Ord k, Eq v)
+         => mk1 k v
+         -> mk2 k v
+         -> f (mk3 k v)
+       )
+    -> LedgerTables l mk1
+    -> LedgerTables l mk2
+    -> f (LedgerTables l mk3)
+
+  zipLedgerTables2A ::
+       Applicative f
+    => (forall k v.
+            (Ord k, Eq v)
+         => mk1 k v
+         -> mk2 k v
+         -> mk3 k v
+         -> f (mk4 k v)
+       )
+    -> LedgerTables l mk1
+    -> LedgerTables l mk2
+    -> LedgerTables l mk3
+    -> f (LedgerTables l mk4)
+
+  foldLedgerTables ::
+       Monoid m
+    => (forall k v.
+            (Ord k, Eq v)
+         => mk k v
+         -> m
+       )
+    -> LedgerTables l mk
+    -> m
+
+  foldLedgerTables2 ::
+       Monoid m
+    => (forall k v.
+           (Ord k, Eq v)
+        => mk1 k v
+        -> mk2 k v
+        -> m
+       )
+    -> LedgerTables l mk1
+    -> LedgerTables l mk2
+    -> m
+
+  namesLedgerTables :: LedgerTables l NameMK
+
+overLedgerTables ::
+     (TableStuff l, IsMapKind mk1, IsMapKind mk2)
+  => (LedgerTables l mk1 -> LedgerTables l mk2)
+  -> l mk1
+  -> l mk2
+overLedgerTables f l = withLedgerTables l $ f $ projectLedgerTables l
+
+mapOverLedgerTables ::
+     (TableStuff l, IsMapKind mk1, IsMapKind mk2)
+  => (forall k v.
+          (Ord k, Eq v)
+       => mk1 k v
+       -> mk2 k v
+     )
+  -> l mk1
+  -> l mk2
+mapOverLedgerTables f = overLedgerTables $ mapLedgerTables f
+
+zipOverLedgerTables ::
+     (TableStuff l, IsMapKind mk1, IsMapKind mk3)
+  => (forall k v.
+          (Ord k, Eq v)
+       => mk1 k v
+       -> mk2 k v
+       -> mk3 k v
+     )
+  ->              l mk1
+  -> LedgerTables l mk2
+  ->              l mk3
+zipOverLedgerTables f l tables2 =
+    overLedgerTables
+      (\tables1 -> zipLedgerTables f tables1 tables2)
+      l
+
+class TableStuff l => TickedTableStuff (l :: LedgerStateKind) where
+  -- | NOTE: The 'IsMapKind mk2' constraint is here for the same reason
+  -- it's on 'projectLedgerTables'
+  projectLedgerTablesTicked :: IsMapKind mk => Ticked1 l mk  -> LedgerTables l mk
+  -- | NOTE: The 'IsMapKind mk2' constraint is here for the same reason
+  -- it's on 'withLedgerTables'
```

Comment:
Same as my previous comment: The constraints could probably be removed, and the comments are outdated.